docs/logql.md

```diff
@@ -0,0 +1,159 @@
+is then used to do a distributed `grep` over the retrieved log streams.
+
+### Log Stream Selector
+
+The log stream selector determines which log streams should be included in your
+query. The stream selector is comprised of one or more key-value pairs, where
+each key is a **log label** and the value is that label's value.
+
+The log stream selector is written by wrapping the key-value pairs in a
+pair of curly braces:
+
+```
+{app="mysql",name="mysql-backup"}
+```
+
+In this example, log streams that have a label of `app` and `name` with exact
+values of `mysql` and `mysql-backup` will be included in the query results.
+
+The `=` operator after the label name is a **label matching operator**. The
+following label matching operators are supported:
+
+- `=`: exactly equal.
+- `!=`: not equal.
+- `=~`: regex matches.
+- `!~`: regex does not match.
+
+Examples:
+
+- `{name=~"mysql.+"}`
+- `{name!~"mysql.+"}`
+
+The same rules that apply for [Prometheus Label
+Selectors](https://prometheus.io/docs/prometheus/latest/querying/basics/#instant-vector-selectors)
+apply for Loki log stream selectors.
+
+### Filter Expression
+
+After writing the log stream selector, the resulting set of logs can be filtered
+further with a search expression. The search expression can be just text or
+regex:
+
+- `{job="mysql"} |= "error"`
+- `{name="kafka"} |~ "tsdb-ops.*io:2003"`
+- `{instance=~"kafka-[23]",name="kafka"} != kafka.server:type=ReplicaManager`
+
+In the previous examples, `|=`, `|~`, and `!=` act as **filter operators** and
+the following filter operators are supported:
+
+- `|=`: Log line contains string.
+- `!=`: Log line does not contain string.
+- `|~`: Log line matches regular expression.
+- `!~`: Log line does not match regular expression.
+
+Filter operators can be chained and will sequentially filter down the
+expression - resulting log lines must satisfy _every_ filter:
+
+`{job="mysql"} |= "error" != "timeout"`
+
+When using `|~` and `!~`,
+[RE2 syntax](https://github.com/google/re2/wiki/Syntax) regex may be used. The
+matching is case-sensitive by default and can be switched to case-insensitive
+prefixing the regex with `(?i)`.
+
+### Future Query Language Extensions
+
+The query language is still under development to support more features, for
+example:
+
+- `AND` / `NOT` operators
+- JSON accessors for filtering of JSON-structured logs
+- Context (like `grep -C n`)
+
+## Counting logs
+
+LogQL also supports functions that wrap a query and allow for counting entries
+per stream.
+
+### Range Vector aggregation
+
+LogQL shares the same [range vector](https://prometheus.io/docs/prometheus/latest/querying/basics/#range-vector-selectors)
+concept from Prometheus, except the selected range of samples include a value of
+1 for each log entry. An aggregation can be applied over the selected range to
+transform it into an instance vector.
+
+`rate` calculates the number of entries per second while `count_over_time`
```

Comment:
I would be more clear is, starting with "The following functions are supported:" (and then a list). Otherwise the reader may ask if `rate` and `count_over_time` are just an example here, or they're the only functions supported.